src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts

```diff
@@ -348,16 +350,19 @@ class adminLogs extends viewModelBase {
         this.allData.push(msg);
         
         if (showMessageNow) {
-            this.listController().pushElements([...this.pendingMessages, msg]);
+            const filteredItems = this.pendingMessages.filter(this.getFilterFunction());
+            this.listController().pushElements([...filteredItems, msg]);
+            this.pendingMessages.length = 0;
+            
         } else {
             this.pendingMessages.push(msg);
         }
     }
     
-    private onAppendPendingMessages() {
+    private onMessageAdded() {
```

Comment:
name is wrong ihmo.

Reply:
we used to just append to `this.pendingMessages`
but now there is also the `showMessageNow `flow
and then we don't append to `this.pendingMesages`, but **add** to `listController`